vignettes/articles/oauth.Rmd

```diff
@@ -13,39 +13,95 @@ knitr::opts_chunk$set(
 library(httr2)
 ```
 
-If the API provides access to a website where the user already has an account (think Twitter, Instagram, Facebook, Google, GitHub, etc), it's likely to use OAuth to allow you to authorise on behalf of the user.
-OAuth[^1] is an authorisation framework that's designed so that you don't have to share your username and password with an app; instead the app asks for permission to use your account.
-You've almost certainly used this before on the web; it's used in most cases where one website wants to use another website on your behalf.
+OAuth[^1] is an authorization framework design for performing work on the behalf of a user.
+You've probably used it a bunch without knowing what it's called: it's used when you login to a non-Google website using your Google account, when you give an phone access to your twitter account, or when you login to a streaming app on your smart TV.
 
 [^1]: Here I'll only talk about OAuth 2.0 which is the only version in common use today.
     OAuth 1.0 is largely only of historical interest.
 
-OAuth is a broad framework that has many many many different variants which makes it hard to provide generalisable advice.
-The following advice draws on my experience working with a number of OAuth using APIs, but don't be surprised if you need to do something slightly different for the API you're working with.
+You'll notice none of these common scenarios quite match to using an R package to scrape data from a web API, and that's the pleasure and pain of OAuth.
+OAuth gives you the incredible power to extract data from popular web services, but it's fundamentally designed for a different use case.
+This means that while httr2 can make the interactive experience pretty seamless, unattended usage (i.e. on CI or in an automated script) is usually going to include some pain.
+
+This vignette builds on the techniques in discussed in `vignette("wrapping-apis")`, so I recommend you read that vignette first.
+I'll also assume that you're working in package, since most people want to wrap up an API into a bunch of R functions, but most of the ideas will also apply if you're creating a one off script.
+
+## OAuth basics
```

Comment:
@llrs @maelle @jonthegeek thanks for all your feedback so far 😄 I added this new section and if you have time to take a look, I'd definitely appreciate your thoughts!